File: getting-started/src/main/java/tech/donau/course/GreetingResource.java
package tech.donau.course;

import tech.donau.course.service.GreetingsService;

import javax.inject.Inject;
import javax.ws.rs.*;
import javax.ws.rs.core.MediaType;

@Path("/hello")
public class GreetingResource {

    private final GreetingsService service;

    @Inject
    public GreetingResource(GreetingsService service) {
        this.service = service;
    }

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    public String hello() {
        return String.format("ID: %s",service.getId());
    }

    @POST
    @Consumes(MediaType.APPLICATION_JSON)
    @Produces(MediaType.APPLICATION_JSON)
    public String postJson(@HeaderParam("token") String token, @QueryParam("name") String name) {
        if (token == null) {
            throw new IllegalArgumentException("Unauthorized");
        }
        return String.format("{\"name\": \"%s\"}", name);
    }

    @PUT
    @Path("/{id}")
    @Produces(MediaType.APPLICATION_JSON)
    public void get(@PathParam("id") String id) {
        System.out.println(id);
    }
}

File: getting-startedk/src/main/kotlin/tech/donau/course/GreetingResource.kt
package tech.donau.course

import tech.donau.course.service.GreetingService
import javax.enterprise.inject.Default
import javax.inject.Inject
import javax.ws.rs.GET
import javax.ws.rs.Path
import javax.ws.rs.PathParam
import javax.ws.rs.Produces
import javax.ws.rs.core.MediaType

@Path("/hello")
class GreetingResource {

    @Inject
    @field: Default
    lateinit var greetingService: GreetingService;

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    fun hello() = "Hello RESTEasy"

    @GET
    @Produces(MediaType.TEXT_PLAIN)
    fun greet(@PathParam("name") name: String): String = greetingService.greetIt(name)
}